src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -424,11 +428,7 @@ private void StartInactiveCursors(StreamId streamId, StreamConsumerCollection st
         {
             try
             {
-                // double check in case of interleaving
-                if (consumerData.State == StreamConsumerDataState.Active ||
-                    consumerData.Cursor == null) return;
-                
-                consumerData.State = StreamConsumerDataState.Active;
+                if (logger.IsVerbose) logger.Verbose("RunConsumerCursor: consumerData={0}.", consumerData);
                 while (consumerData.Cursor != null && consumerData.Cursor.MoveNext())
```

Comment:
We used to move the cursor to Next, which grabs the last item and moves the cursor to point to null (internally cursor will be at state !IsSet), which means the bucketing system in the cache thins this cursor is not active.
Next we go ad deliver that msgs to the grain, await it and only after that we do `consumerData.State = StreamConsumerDataState.Inactive;`.
But in the meanwhile, if new msg arrives, it will miss this cursor and will also not wake up this cursor, since this cursor is still not in Inactive state.